src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -788,7 +788,8 @@ static int client_parse_message(
                 uint8_t *optval;
                 be32_t iaid_lease;
 
-                if (len < sizeof(DHCP6Option) || len < sizeof(DHCP6Option) + be16toh(option->len))
+                if (len < offsetof(DHCP6Option, data) ||
+                    len < offsetof(DHCP6Option, data) + be16toh(option->len))
```

Comment:
btw, my comment was not exclusive to this part of the code, the same applies to a number of other cases too when handling DHCP6Option

Reply:
And I didn't really have my eyes open to the other cases, either. Sorry for that. Stashed the task on a yellow sticker to be done later.